### PR TITLE
Don’t offer import statement completions at `from` position

### PR DESCRIPTION
**Pull request by andrewbranch**
❗ Hoping to get this fix into for 4.3. The current state is pretty disruptive if your muscle memory expects a completion for `from` but you get an import completion instead.

Fixes this:

import statement completions showing auto imports starting with 'f' at 'import {} f'

By making it this:

completions at 'f' now shows 'from' and no auto imports

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -183,6 +185,20 @@ namespace ts.Completions {
         return { isGlobalCompletion: false, isMemberCompletion: false, isNewIdentifierLocation: false, entries };
     }
 
+    function specificKeywordCompletionInfo(keywords: readonly SyntaxKind[]): CompletionInfo {
```
Comment (andrewbranch) [inline src/services/completions.ts:188]: Making a way to return specific keywords is extra credit beyond just fixing the obvious regression; the behavior in 4.2 was not great:

image

File: src/services/completions.ts [hunk 2]
```diff
@@ -183,6 +185,20 @@ namespace ts.Completions {
         return { isGlobalCompletion: false, isMemberCompletion: false, isNewIdentifierLocation: false, entries };
     }
 
+    function specificKeywordCompletionInfo(keywords: readonly SyntaxKind[]): CompletionInfo {
+        return {
+            isGlobalCompletion: true,
```
Comment (andrewbranch) [inline src/services/completions.ts:190]: I tried making this `false` but the snippets for `for` loops and stuff still showed up 🤔

**Review by sheetalkamat — APPROVED** [review]

File: src/services/completions.ts (hunk 2 above)
Comment (DanielRosenwasser) [reply src/services/completions.ts:190]: Should it be `false` then?

@mjbvz any idea why that might be happening?

File: src/services/completions.ts [hunk 3]
```diff
@@ -802,6 +818,8 @@ namespace ts.Completions {
                         return JsDoc.getJSDocTagCompletionDetails(name);
                     case CompletionDataKind.JsDocParameterName:
                         return JsDoc.getJSDocParameterNameCompletionDetails(name);
+                    case CompletionDataKind.Keywords:
+                        return request.keywords.some(k => tokenToString(k) === name) ? createSimpleDetails(name, ScriptElementKind.keyword, SymbolDisplayPartKind.keyword) : undefined;
```
Comment (DanielRosenwasser) [inline src/services/completions.ts:822]: Maybe instead of this, try looking up the `SyntaxKind` for `name` and just do `indexOf`.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
Seems right, but I feel like you should specify `isGlobalCompletions` as `false` (or at least get clarity on behavior), and you could also just switch to a cheaper lookup for `name`.

Once that's done, we can cherry-pick into release-4.3.

File: src/services/completions.ts (hunk 2 above)
Comment (andrewbranch) [reply src/services/completions.ts]: Set to `false` now despite seeing no difference locally

**Comment by andrewbranch** [comment]
@typescript-bot cherry-pick this to release-4.3

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**